app/src/main/java/moe/apex/rule34/DeepLinkActivity.kt

```diff
@@ -55,8 +57,16 @@ class DeepLinkActivity : SingletonImageLoader.Factory, ComponentActivity() {
 
         setContent {
             val prefs = prefs.getPreferences.collectAsState(initialPrefs).value
+            val uri = mutableStateOf(intent.data)
```

Comment:
Wrap the mutable state in `rememberSaveable` otherwise configuration changes (like rotating the screen) will cause the previous/original image to appear again.